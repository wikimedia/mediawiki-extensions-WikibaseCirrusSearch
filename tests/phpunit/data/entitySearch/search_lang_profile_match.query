{
	"search": "Abr",
	"language": "de",
	"userLang": "de",
	"type": "item",
	"strictlanguage": false,
	"profiles": {
		"wikibase_prefix_querybuilder": {
			"default-de": {
				"language-chain": ["de", "en"],
				"tie-breaker": 0.2,
				"de-exact": 9,
				"de-folded": 8,
				"de-prefix": 7,
				"en-exact": 6,
				"en-folded": 5,
				"en-prefix": 4,
				"any": 1,
				"space-discount": 1
			}
		}
	}
}
