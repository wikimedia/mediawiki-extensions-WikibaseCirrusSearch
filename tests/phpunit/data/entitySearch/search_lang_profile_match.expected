{
    "description": "wikibase_prefix search for 'Abr'",
    "params": {
        "timeout": "20s"
    },
    "query": {
        "query": {
            "bool": {
                "should": [
                    {
                        "bool": {
                            "filter": [
                                {
                                    "match": {
                                        "labels_all.prefix": "Abr"
                                    }
                                }
                            ],
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 0.2,
                                        "queries": [
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.near_match": "Abr"
                                                        }
                                                    },
                                                    "boost": 9
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.near_match_folded": "Abr"
                                                        }
                                                    },
                                                    "boost": 8
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.de.prefix": "Abr"
                                                        }
                                                    },
                                                    "boost": 7
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels_all.near_match_folded": "Abr"
                                                        }
                                                    },
                                                    "boost": 1
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match": "Abr"
                                                        }
                                                    },
                                                    "boost": 6
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.near_match_folded": "Abr"
                                                        }
                                                    },
                                                    "boost": 5
                                                }
                                            },
                                            {
                                                "constant_score": {
                                                    "filter": {
                                                        "match": {
                                                            "labels.en.prefix": "Abr"
                                                        }
                                                    },
                                                    "boost": 4
                                                }
                                            }
                                        ]
                                    }
                                }
                            ]
                        }
                    },
                    {
                        "term": {
                            "title.keyword": "ABR"
                        }
                    }
                ],
                "minimum_should_match": 1,
                "filter": [
                    {
                        "term": {
                            "content_model": "wikibase-item"
                        }
                    }
                ]
            }
        },
        "_source": [
            "namespace",
            "title",
            "namespace_text",
            "wiki",
            "labels.de",
            "descriptions.de",
            "labels.en",
            "descriptions.en"
        ],
        "fields": [],
        "highlight": {
            "pre_tags": [
                ""
            ],
            "post_tags": [
                ""
            ],
            "fields": {
                "title": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "number_of_fragments": 0,
                    "matched_fields": [
                        "title.keyword"
                    ]
                },
                "labels.de.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "order": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true,
                        "return_snippets_and_offsets": true
                    }
                },
                "labels.en.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "order": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true,
                        "return_snippets_and_offsets": true
                    }
                },
                "labels.*.prefix": {
                    "type": "experimental",
                    "fragmenter": "none",
                    "order": "none",
                    "number_of_fragments": 0,
                    "options": {
                        "skip_if_last_matched": true,
                        "return_snippets_and_offsets": true
                    }
                }
            }
        },
        "size": 10,
        "rescore": [
            {
                "window_size": 8192,
                "query": {
                    "query_weight": 1,
                    "rescore_query_weight": 1,
                    "score_mode": "total",
                    "rescore_query": {
                        "function_score": {
                            "score_mode": "sum",
                            "functions": [
                                {
                                    "script_score": {
                                        "script": {
                                            "source": "pow(doc['incoming_links'].value , 1) \/ ( pow(doc['incoming_links'].value, 1) + pow(100,1))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 0.6
                                },
                                {
                                    "script_score": {
                                        "script": {
                                            "source": "pow(doc['sitelink_count'].value , 2) \/ ( pow(doc['sitelink_count'].value, 2) + pow(20,2))",
                                            "lang": "expression"
                                        }
                                    },
                                    "weight": 0.4
                                }
                            ]
                        }
                    }
                }
            }
        ],
        "stats": [
            "wikibase-prefix"
        ]
    },
    "options": {
        "timeout": "20s"
    }
}